cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3833,19 +3840,21 @@
 
 ;; Helper for creating `subps` instructions.
 (decl x64_subps (Xmm XmmMem) Xmm)
-(rule 0 (x64_subps src1 src2)
-      (xmm_rm_r (SseOpcode.Subps) src1 src2))
-(rule 1 (x64_subps src1 src2)
+(rule 2 (x64_subps src1 src2)
       (if-let true (use_avx))
       (xmm_rmir_vex (AvxOpcode.Vsubps) src1 src2))
+(rule 1 (x64_subps src1 (is_xmm_mem src2)) (x64_subps_a src1 src2))
+(rule 0 (x64_subps src1 src2)
+      (xmm_rm_r (SseOpcode.Subps) src1 src2))
 
 ;; Helper for creating `subpd` instructions.
 (decl x64_subpd (Xmm XmmMem) Xmm)
-(rule 0 (x64_subpd src1 src2)
-      (xmm_rm_r (SseOpcode.Subpd) src1 src2))
-(rule 1 (x64_subpd src1 src2)
+(rule 2 (x64_subpd src1 src2)
       (if-let true (use_avx))
       (xmm_rmir_vex (AvxOpcode.Vsubpd) src1 src2))
+(rule 1 (x64_subpd src1 (is_xmm_mem src2)) (x64_subpd_a src1 src2))
+(rule 0 (x64_subpd src1 src2)
+      (xmm_rm_r (SseOpcode.Subpd) src1 src2))
```

Comment:
Let's remove the old emission. In a recent PR I began removing these for scalar instructions. It ensures that our `is_xmm_mem` is operating as expected.